src/main/scala/is/hail/expr/Relational.scala

```diff
@@ -334,6 +364,56 @@ case class FilterCols(
   }
 }
 
+case class FilterColsIR(
+  child: MatrixIR,
+  pred: IR) extends MatrixIR {
+
+  def children: IndexedSeq[BaseIR] = Array(child, pred)
+
+  def copy(newChildren: IndexedSeq[BaseIR]): FilterColsIR = {
+    assert(newChildren.length == 2)
+    FilterColsIR(newChildren(0).asInstanceOf[MatrixIR], newChildren(1).asInstanceOf[IR])
+  }
+
+  def typ: MatrixType = child.typ
+
+  def execute(hc: HailContext): MatrixValue = {
+    val prev = child.execute(hc)
+
+    val localGlobals = prev.globals.broadcast
+    val localColType = typ.colType
+    val ec = typ.colEC
+    
+    //
+    // Initialize a region containing the globals
+    //
+    val colRegion = Region()
+    val rvb = new RegionValueBuilder(colRegion)
+    rvb.start(typ.globalType)
+    rvb.addAnnotation(typ.globalType, localGlobals.value)
+    val globalRVend = rvb.currentOffset()
+    val globalRVoffset = rvb.end()
+
+    val (rTyp, predCompiledFunc) = ir.Compile[Long, Long, Boolean](
+      "global", typ.globalType,
+      "sa",     typ.colType,
+      pred
+    )
+    // We don't yet support IR aggregators
+    assert(ec.aggregations.isEmpty)
```

Comment:
`EvalContext` is for compiling AST, but since this code is IR-based, I don't think it should be mentioned.  In particular, there's no reason to construct the `ec` by calling `colEC`, `ec.aggregations` is populated during `AST.compile` but you're not calling that here (so it is necessarily empty), and `ec` is unused below so it is dead.